reframe/utility/sanity.py

```diff
@@ -530,28 +531,65 @@ def extractiter(patt, filename, tag=0, conv=None, encoding='utf-8'):
     the extracted values.
     '''
     for m in finditer(patt, filename, encoding):
-        try:
-            val = m.group(tag)
-        except (IndexError, KeyError):
-            raise SanityError(
-                "no such group in pattern `%s': %s" % (patt, tag))
-
-        try:
-            yield conv(val) if callable(conv) else val
-        except ValueError:
-            fn_name = '<unknown>'
+        if isinstance(tag, collections.Iterable) and not isinstance(tag, str):
```

Comment:
Don't do everything inline. Delegate to different functions for each case.

Reply:
It's better if you delegate to separate function completely in case `tag` is an iterable, and do that right at the beginning:

```python
    if isinstance(tag, collections.Iterable) and not isinstance(tag, str):
        return _extractiter_multitag(patt, filename, tag, conv, encoding)

    # And here remains are as before
```